ui/component/or-chart/src/index.ts

```diff
@@ -984,44 +1050,54 @@ export class OrChart extends translate(i18next)(LitElement) {
+                let dataset = await this._loadAttributeData(asset, attribute, this.colors[colourIndex], interval, this._startOfPeriod!, this._endOfPeriod!, false, asset.name + " " + label);
+                (dataset as any).assetId = asset.id;
+                (dataset as any).attrName = attribute.name;
+                (dataset as any).unit = unit;
                 data.push(dataset);
 
-                predictedFromTimestamp = now < startOfPeriod ? startOfPeriod : now;
-                dataset = await this._loadAttributeData(this.assets[assetIndex], attribute, this.colors[colourIndex], interval, startOfPeriod, endOfPeriod, true,  asset.name + " " + label + " " + i18next.t("compare") + " " + i18next.t("predicted"));
-                dataset.data.forEach((dp) => dp.x += offset);
-                dataset.borderDash = [6, 8];
-                (dataset as any).isComparisonDataset = true;
+                dataset =  await this._loadAttributeData(this.assets[assetIndex], attribute, this.colors[colourIndex], interval, predictedFromTimestamp, this._endOfPeriod!, true, asset.name + " " + label + " " + i18next.t("predicted"));
                 data.push(dataset);
-            }
-        });
+
+                if (this.compareTimestamp) {
+                    const startOfPeriod = moment(this.compareTimestamp).startOf(this.period!).startOf(lowerCaseInterval as moment.unitOfTime.StartOf).add(1, lowerCaseInterval as moment.unitOfTime.Base).toDate().getTime();
+                    const endOfPeriod = moment(this.compareTimestamp).endOf(this.period!).startOf(lowerCaseInterval as moment.unitOfTime.StartOf).add(1, lowerCaseInterval as moment.unitOfTime.Base).toDate().getTime();
+                    const offset =  this._startOfPeriod! - startOfPeriod;
+
+                    dataset = await this._loadAttributeData(this.assets[assetIndex], attribute, this.colors[colourIndex], interval, startOfPeriod, endOfPeriod, false,  asset.name + " " + label + " " + i18next.t("compare"));
+                    dataset.data.forEach((dp) => dp.x += offset);
+                    dataset.borderDash = [10, 10];
+                    (dataset as any).isComparisonDataset = true;
+                    data.push(dataset);
+
+                    predictedFromTimestamp = now < startOfPeriod ? startOfPeriod : now;
+                    dataset = await this._loadAttributeData(this.assets[assetIndex], attribute, this.colors[colourIndex], interval, startOfPeriod, endOfPeriod, true,  asset.name + " " + label + " " + i18next.t("compare") + " " + i18next.t("predicted"));
+                    dataset.data.forEach((dp) => dp.x += offset);
+                    dataset.borderDash = [6, 8];
+                    (dataset as any).isComparisonDataset = true;
+                    data.push(dataset);
+                }
+            });
+        }
 
         this._loading = false;
-        await Promise.all(promises);
+        if(promises) {
+            await Promise.all(promises);
+        }
```

Comment:
Looks like this should be moved inside the else statement so the `_loading` variables is only set to false once the promises complete

Reply:
Seems that the `await` function was meant to be started after `_loading` was set to false.
Just tried to switch them around, but then it will only show data points after a second component render.
It was set in this order beforehand, so I'll just keep it like this I guess.